src/lib/utils.ts

```diff
@@ -37,29 +39,124 @@ export function promisifyServiceClient(client: Client) {
   });
 }
 
-interface FieldMask {
-  pathsList: string[];
+function isValueField(isObject: boolean, value: any): boolean {
+  if (typeof value === "undefined") {
+    return false;
+  }
+  return isObject ? value.hasOwnProperty("value") && Object.keys(value).length === 1 : false;
 }
 
-export function formatCallResults(resultsList: any[], fieldMask: FieldMask | undefined) {
-  const parsedResults: any[] = [];
+function getNestedValue(path: string, data: GoogleAdsRow): any {
+  const value = get(data, path) ?? get(data, `${path}List`);
+  if (typeof value === "undefined") {
+    return value;
+  }
+
+  // Convert UNKNOWN enum values to UNSPECIFIED
+  const isEnum = typeof value === "number";
+  if (isEnum && value === UNKNOWN_ENUM_VALUE) {
```

Comment:
Another thing here: I don't think this fixes the problem with unknown enums.

The problem arises when an enum comes up (such as `17`) which isn't defined in our protos. So later, when we try to do a mutation with that enum set to `17`, GRPC throws an error.

The correct behavior here would be to detect whether `17` is a valid enum according to the protos, and to set it to `1` if not. @Midnight-Conqueror You should have an account to test this on, with the `conversionactioncategory` enum.

Reply:
Hmm interesting, a bit harder than I thought then. I'm going to remove this for now then, since I think it will be quite challenging (we can do it another time).